src/SAML2/XML/ds/KeyInfo.php

```diff
@@ -31,45 +30,19 @@ class KeyInfo
      *
      * @var (\SAML2\XML\Chunk|\SAML2\XML\ds\KeyName|\SAML2\XML\ds\X509Data)[]
      */
-    private $info = [];
+    protected $info = [];
 
 
     /**
      * Initialize a KeyInfo element.
      *
-     * @param \DOMElement|null $xml The XML element we should load.
+     * @param (\SAML2\XML\Chunk|\SAML2\XML\ds\KeyName|\SAML2\XML\ds\X509Data)[] $info
+     * @param string|null $Id
      */
-    public function __construct(DOMElement $xml = null)
+    public function __construct(array $info, $Id = null)
```

Comment:
Shouldn't we specify the `string` type hint for `$Id` here?

Reply:
```
<element name="KeyInfo" type="ds:KeyInfoType"/>
- <complexType name="KeyInfoType" mixed="true">
- <choice maxOccurs="unbounded">
<element ref="ds:KeyName"/>
<element ref="ds:KeyValue"/>
<element ref="ds:RetrievalMethod"/>
<element ref="ds:X509Data"/>
<element ref="ds:PGPData"/>
<element ref="ds:SPKIData"/>
<element ref="ds:MgmtData"/>
<any processContents="lax" namespace="##other"/>
<!-- (1,1) elements from (0,unbounded) namespaces -->
</choice>
<attribute name="Id" type="ID" use="optional"/>
</complexType>
```

Reply:
The difference between `xsd:ID` and `xsd:string` is only semantical, if I'm not mistaken. In practice, an `xsd:ID` would always translate into a string for us.